compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -253,15 +263,47 @@ impl Collector<'tcx> {
             )
             .emit();
         }
-        if lib.kind == NativeLibKind::RawDylib && !self.tcx.features().raw_dylib {
-            feature_err(
-                &self.tcx.sess.parse_sess,
-                sym::raw_dylib,
-                span.unwrap_or(rustc_span::DUMMY_SP),
-                "kind=\"raw-dylib\" is unstable",
-            )
-            .emit();
+        // this just unwraps lib.name; we already established that it isn't empty above.
+        if let (NativeLibKind::RawDylib, Some(lib_name)) = (lib.kind, lib.name) {
+            let span = match span {
+                Some(s) => s,
+                None => {
+                    // Disallow this case: raw-dylib is not useful without function signatures,
+                    // because we need those in order to generate the import library, and users
+                    // cannot specify functions on the command line.
+                    self.tcx
+                        .sess
+                        .fatal("raw-dylib libraries are not supported on the command line");
```

Comment:
This case should be impossible, `raw-dylib` kind is rejected by option parser, so we never get here.
You can add a test case for it to `src/test/ui/manual/manual-link-bad-kind.rs`.